Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -290,38 +313,35 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
 void CDMInstanceOpenCDM::updateLicense(const String& sessionId, LicenseType, const SharedBuffer& response, LicenseUpdateCallback callback)
 {
     std::string responseMessage;
-    // FIXME: This is a realy awful API, why not just take a string& ?
-    int haveMessage = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
-    GST_DEBUG("session id %s, calling callback %s message", sessionId.utf8().data(), haveMessage ? "with" : "without");
-
-    if (haveMessage) {
-        GST_TRACE("OpenCDM::Update has a message for us");
+    media::OpenCdm::KeyStatus keyStatus = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
+    GST_DEBUG("session id %s, key status is %ld", sessionId.utf8().data(), static_cast<long>(keyStatus));
+    if (keyStatus == media::OpenCdm::KeyStatus::Usable) {
+        SharedBuffer* initData = sessionIdMap.get(sessionId);
+        KeyStatusVector changedKeys;
+        GST_TRACE("OpenCDM::update returned key is usable");
+        // FIXME: Why are we passing initData here, we are supposed to be passing key IDs.
+        changedKeys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus> {*initData, openCDMKeyStatusToWebKit(keyStatus)});
+        callback(false, WTFMove(changedKeys), std::nullopt, std::nullopt, SuccessValue::Succeeded);
+    } else if (keyStatus != media::OpenCdm::KeyStatus::InternalError) {
```

Comment:
We callback with succeeded, but the key status could be released or expired? Doesn't that imply we would need to run the Update Key Statuses algo?

Reply:
It might, we could add it to the FIXME.